### PR TITLE
Fix issue #6670.  Avoid infinite recursion from TClass::InheritsFrom.

### PR DESCRIPTION
**Pull request by pcanal**
Do not inadvertently request TClass::fProperty's initialization during TProtoClass::FillTClass,
the former ends up calling TProtoClass::FillTClass thus leading to an infinite recursion.

### CODE REVIEW

File: core/meta/src/TProtoClass.cxx [hunk 1]
```diff
@@ -175,7 +175,7 @@ Bool_t TProtoClass::FillTClass(TClass* cl) {
             Info("FillTClass", "Returning w/o doing anything. %s is a STL collection.",cl->GetName());
          return kFALSE;
       }
-      if (cl->Property() & kIsNamespace) {
+      if (cl->fProperty != (-1) && cl->fProperty & kIsNamespace) {
```
Comment (jblomer) [inline core/meta/src/TProtoClass.cxx:178]: Is there somewhere a documentation of the meaning of the possible `fProperty` values?

**Review by jblomer — CHANGES_REQUESTED** [review]

File: core/meta/src/TProtoClass.cxx (hunk 1 above)
Comment (Axel-Naumann) [inline core/meta/src/TProtoClass.cxx:178]:
```suggestion
      if (cl->fProperty != -1 && (cl->fProperty & kIsNamespace)) {
```

Comment (Axel-Naumann) [reply core/meta/src/TProtoClass.cxx:178]: `0` isn't a thing for `fProperty`, and isn't tested for in the original code or Philippe's modification. `fProperty & kIsWhatever` checks whether the class is a whatever. `-1` is the default value of `fProperty`, stating "we have not determined the class's properties yet" (i.e. `Property()` has not been called yet)

Comment (pcanal) [reply core/meta/src/TProtoClass.cxx:178]: `-1` means unset.  0 would means all properties was found to be 'false'.
Yes, I need to update the documentation :)

File: core/meta/src/TClass.cxx [hunk 2]
```diff
@@ -5892,21 +5892,28 @@ void TClass::PostLoadCheck()
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-/// Set TObject::fBits and fStreamerType to cache information about the
-/// class.  The bits are
+/// Returns the properties of the TClass a bit field stored as a `Long_t` value.
```
Comment (Axel-Naumann) [inline core/meta/src/TClass.cxx:5895]:
```suggestion
/// Returns the properties of the TClass as a bit field stored as a `Long_t` value.
```

**Review by Axel-Naumann — APPROVED** [review]

**Review by jblomer — APPROVED** [review]

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by pcanal** [comment]
The failures are unrelated.

**Pull request merged**